File: ios/Sources/VaultSync/Domain/Repositories/SyncRepository.swift
import Foundation

public protocol SyncRepository: Sendable {
    func observePendingOperations() -> AsyncStream<[SyncOperation]>
    func getPendingOperations() async throws -> [SyncOperation]
    func getRunningOperations() async throws -> [SyncOperation]
    func recordOperation(_ operation: SyncOperation) async throws
    func updateOperationStatus(
        id: String,
        status: OperationStatus,
        errorMessage: String?,
        completedAt: Date?
    ) async throws
    func incrementRetryCount(id: String) async throws
    func getConflicts() async throws -> [Conflict]
    func observeConflicts() -> AsyncStream<[Conflict]>
    func recordConflict(_ conflict: Conflict) async throws
    func resolveConflict(conflictId: String, resolution: ConflictResolution) async throws
}

File: android/app/src/main/java/com/vaultsync/domain/repository/SyncRepository.kt
package com.vaultsync.domain.repository

import com.vaultsync.domain.model.Conflict
import com.vaultsync.domain.model.ConflictResolution
import com.vaultsync.domain.model.OperationStatus
import com.vaultsync.domain.model.SyncOperation
import kotlinx.coroutines.flow.Flow

interface SyncRepository {
    fun observePendingOperations(): Flow<List<SyncOperation>>
    suspend fun getPendingOperations(): List<SyncOperation>
    suspend fun getRunningOperations(): List<SyncOperation>
    suspend fun recordOperation(operation: SyncOperation)
    suspend fun updateOperationStatus(
        id: String,
        status: OperationStatus,
        errorMessage: String? = null,
        completedAt: Long? = null
    )
    suspend fun incrementRetryCount(id: String)
    suspend fun getConflicts(): List<Conflict>
    fun observeConflicts(): Flow<List<Conflict>>
    suspend fun recordConflict(conflict: Conflict)
    suspend fun resolveConflict(conflictId: String, resolution: ConflictResolution)
}
